web-admin/src/features/scheduled-reports/query-mapper.ts

```diff
@@ -0,0 +1,277 @@
+  const req: ReportQueryRequest = correctRequest(
+    JSON.parse(reportResource.report.spec.queryArgsJson),
+  );
+  let queryMapper: (args: QueryMapperArgs<ReportQueryRequest>) => void;
+
+  // get metrics view name and the query mapper function based on the query name.
+  switch (reportResource.report.spec.queryName) {
+    case "MetricsViewAggregation":
+      metricsViewName = (req as V1MetricsViewAggregationRequest).metricsView;
+      queryMapper = mapMetricsViewAggregationRequest;
+      break;
+    case "MetricsViewToplist":
+      metricsViewName = (req as V1MetricsViewToplistRequest).metricsViewName;
+      queryMapper = mapMetricsViewToplistRequest;
+      break;
+    case "MetricsViewRows":
+      metricsViewName = (req as V1MetricsViewRowsRequest).metricsViewName;
+      queryMapper = mapMetricsViewRowsRequest;
+      break;
+    case "MetricsViewTimeSeries":
+      metricsViewName = (req as V1MetricsViewTimeSeriesRequest).metricsViewName;
+      queryMapper = mapMetricsViewTimeSeriesRequest;
+      break;
+    case "MetricsViewComparison":
+      metricsViewName = (req as V1MetricsViewComparisonRequest).metricsViewName;
+      queryMapper = mapMetricsViewComparisonRequest;
+      break;
+  }
+
+  if (!metricsViewName) {
+    // error state
+    return readable({
+      ready: true,
+      error:
+        "Failed to find metrics view name. Please check the format of the report.",
+    });
+  }
+
+  return derived(
+    [
+      useMetaQuery(get(runtime).instanceId, metricsViewName),
+      // TODO: handle non-timestamp dashboards
+      createQueryServiceMetricsViewTimeRange(
+        get(runtime).instanceId,
+        metricsViewName,
+        {},
+      ),
+    ],
+    ([metricsViewResource, timeRangeSummary]) => {
+      if (!metricsViewResource.data || !timeRangeSummary.data)
+        return {
+          ready: false,
+        };
+
+      if (metricsViewResource.error || timeRangeSummary.error) {
+        // error state
+        return {
+          ready: true,
+          error:
+            metricsViewResource.error?.message ??
+            timeRangeSummary.error?.message,
+        };
+      }
+
+      initLocalUserPreferenceStore(metricsViewName);
+      const dashboard = getDefaultMetricsExplorerEntity(
+        metricsViewName,
+        metricsViewResource.data,
+        timeRangeSummary.data,
+      );
+      queryMapper({
+        dashboard,
+        req,
+        metricsView: metricsViewResource.data,
+        timeRangeSummary: timeRangeSummary.data.timeRangeSummary,
+        executionTime,
+      });
+      console.log(dashboard);
+      return {
+        ready: true,
+        state: getProtoFromDashboardState(dashboard),
+        metricsView: metricsViewName,
+      };
+    },
+  );
+}
+
+/**
+ * This method corrects the underscore naming to camel case.
+ * This is the drawback of storing the request object as is.
+ */
+function correctRequest(req: Record<string, any>): Record<string, any> {
```

Comment:
The comment is helpful, but maybe even the function name could be `convertSnakeToCamelCase()`